### ISSUE TITLE
bug(node): Cannot find module '@lancedb/vectordb-linux-arm64-musl'

### ISSUE THREAD

**Issue by nathan-dopko** [lancedb/lancedb#1388]
### What happened?

Im trying to deploy my app which is running on linux and node gives me this error 

Cannot find module '@lancedb/vectordb-linux-arm64-musl'

I noticed this is not a module. Im also not using this anywhere in the code. I cmd+shift+f and this is not in the package.json or anywhere in my code.

here is the proper errors:

June 19, 2024 at 15:37 (UTC-4:00) | throw new Error(`vectordb: failed to load native library. | 86d1f2ff7ca147449f04864a1b6d3887 | outseek-backend
-- | -- | -- | --
June 19, 2024 at 15:37 (UTC-4:00) | ^ | 86d1f2ff7ca147449f04864a1b6d3887 | outseek-backend
June 19, 2024 at 15:37 (UTC-4:00) | Error: vectordb: failed to load native library. | 86d1f2ff7ca147449f04864a1b6d3887 | outseek-backend
June 19, 2024 at 15:37 (UTC-4:00) | You may need to run `npm install @lancedb/vectordb-linux-arm64-musl`. | 86d1f2ff7ca147449f04864a1b6d3887 | outseek-backend
June 19, 2024 at 15:37 (UTC-4:00) | If that does not work, please file a bug report at https://github.com/lancedb/lancedb/issues | 86d1f2ff7ca147449f04864a1b6d3887 | outseek-backend
June 19, 2024 at 15:37 (UTC-4:00) |   | 86d1f2ff7ca147449f04864a1b6d3887 | outseek-backend
June 19, 2024 at 15:37 (UTC-4:00) | Source error: Error: Cannot find module '@lancedb/vectordb-linux-arm64-musl'

June 19, 2024 at 15:37 (UTC-4:00)
throw new Error(`vectordb: failed to load native library.
[86d1f2ff7ca147449f04864a1b6d3887](https://us-east-1.console.aws.amazon.com/ecs/v2/clusters/public-vpc-ECSCluster-mMlyQTXRaSll/services/outseek-backend/tasks/86d1f2ff7ca147449f04864a1b6d3887?region=us-east-1)
outseek-backend
June 19, 2024 at 15:37 (UTC-4:00)
^
[86d1f2ff7ca147449f04864a1b6d3887](https://us-east-1.console.aws.amazon.com/ecs/v2/clusters/public-vpc-ECSCluster-mMlyQTXRaSll/services/outseek-backend/tasks/86d1f2ff7ca147449f04864a1b6d3887?region=us-east-1)
outseek-backend
June 19, 2024 at 15:37 (UTC-4:00)
Error: vectordb: failed to load native library.
[86d1f2ff7ca147449f04864a1b6d3887](https://us-east-1.console.aws.amazon.com/ecs/v2/clusters/public-vpc-ECSCluster-mMlyQTXRaSll/services/outseek-backend/tasks/86d1f2ff7ca147449f04864a1b6d3887?region=us-east-1)
outseek-backend
June 19, 2024 at 15:37 (UTC-4:00)
You may need to run `npm install @lancedb/vectordb-linux-arm64-musl`.
[86d1f2ff7ca147449f04864a1b6d3887](https://us-east-1.console.aws.amazon.com/ecs/v2/clusters/public-vpc-ECSCluster-mMlyQTXRaSll/services/outseek-backend/tasks/86d1f2ff7ca147449f04864a1b6d3887?region=us-east-1)
outseek-backend
June 19, 2024 at 15:37 (UTC-4:00)
If that does not work, please file a bug report at https://github.com/lancedb/lancedb/issues
[86d1f2ff7ca147449f04864a1b6d3887](https://us-east-1.console.aws.amazon.com/ecs/v2/clusters/public-vpc-ECSCluster-mMlyQTXRaSll/services/outseek-backend/tasks/86d1f2ff7ca147449f04864a1b6d3887?region=us-east-1)
outseek-backend
June 19, 2024 at 15:37 (UTC-4:00)
[86d1f2ff7ca147449f04864a1b6d3887](https://us-east-1.console.aws.amazon.com/ecs/v2/clusters/public-vpc-ECSCluster-mMlyQTXRaSll/services/outseek-backend/tasks/86d1f2ff7ca147449f04864a1b6d3887?region=us-east-1)
outseek-backend
June 19, 2024 at 15:37 (UTC-4:00)
Source error: Error: Cannot find module '@lancedb/vectordb-linux-arm64-musl'

**Comment by wjones127** [comment]
Hello, thanks for trying LanceDB.

LanceDB is written in Rust, which is compiled to native code that is specific to your CPU. When you install LanceDB for node, it chooses an appropriate implementation package for your platform. Since your platform is Linux arm64 musl, it tries to download `vectordb-linux-arm64-musl`. But we don't yet build for that platform, so it isn't able to find that package.

TLDR: we don't currently support musl-based operating systems (such as Alpine Linux) for LanceDB. Are you able to use a glibc OS instead, such as Ubuntu?

**Comment by nathan-dopko** [comment]
Hey, @wjones127 thanks for the response. Is their an ETA on building for this platform im trying to use? Or is it better to just try and switch OSs.

**Comment by wjones127** [comment]
> Is there an ETA on building for this platform im trying to use?

We have no timeline to support that platform right now. If someone wanted to submit a PR, we'd be happy to take a look though.

In the near term, I'd recommend using a supported platform.
